### PR TITLE
Added best seller & footer

### LINKED ISSUE #6: Added best seller and footer to landing page
Added best seller books
Created context for product
Added footer to landing page

### PR DESCRIPTION
**Pull request by Raghav888**
Added best seller books
Created context for product
Added footer to landing page

### CODE REVIEW

File: src/components/categorybanner/CategoryBanner.js [hunk 1]
```diff
@@ -1,34 +1,37 @@
...
-        <div key={item.id} className="display-image">
-            <a href="##">
-                <img className="book-type" src={item.categoryImage} alt="love-story" />
+  return (
+    <div>
+      <div className="product-line">
+        {category.map((item) => {
+          return (
+            <div key={item.id} className="display-image">
+              <a href="##">
```
Comment (JantuDeb) [inline src/components/categorybanner/CategoryBanner.js:14]: use `Link` instead of `a`

File: src/components/categorybanner/CategoryBanner.js [hunk 2]
```diff
@@ -1,34 +1,37 @@
...
-        
-    </div>
-    <div className="baner">
+              </a>
+            </div>
+          );
+        })}
+      </div>
+      <div className="baner">
         <a href="##">
```
Comment (JantuDeb) [inline src/components/categorybanner/CategoryBanner.js:27]: same here

File: src/components/shared/navbar/Navbar.js [hunk 3]
```diff
@@ -1,46 +1,51 @@
...
-    <div className="mantra-nav-title">
-        <a className="mantra-title" href="index.html"> BookStore </a>
-    </div>
-    <div className="mantra-nav-search">
-        <input className="mantra-search-box mantra-textbox-classic mantra-highlight-box" placeholder="  Search" type="text " />
+      <div className="mantra-nav-title">
+        <a className="mantra-title" href="index.html">
+          {" "}
+          BookStore{" "}
+        </a>
```
Comment (JantuDeb) [inline src/components/shared/navbar/Navbar.js]: remove empty braces

File: src/context/category-context.js [hunk 4]
```diff
@@ -1,28 +1,30 @@
...
-    return(<CategoryContext.Provider value={{category}}>{children}</CategoryContext.Provider>)
-}
-
-const useCategory=()=>useContext(CategoryContext)
-
-export{useCategory,CategoryProvider}
\ No newline at end of file
+      try {
+        const response = await axios.get("/api/categories");
+        setCategory(response.data.categories);
```
Comment (JantuDeb) [inline src/context/category-context.js:14]: const {data}= await axios.get("/api/categories");

File: src/context/product-listing-context.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  );
+  console.log(productListstate.productList);
+  useEffect(() => {
+    (async () => {
+      try {
+        const response = await axios.get(api);
+        productListdispatch({
+          type: "STATE_UPDATE",
+          payload: { value: response.data.products },
+        });
```
Comment (JantuDeb) [inline src/context/product-listing-context.js]: The `action-type` should be more specific. `ADD_PRODUCTS` would be a good name.

File: src/context/product-listing-context.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+const ProductProvider = ({ children }) => {
+  const [productListstate, productListdispatch] = useReducer(
+    productListReducer,
+    { productList: [] }
+  );
+  console.log(productListstate.productList);
+  useEffect(() => {
+    (async () => {
+      try {
+        const response = await axios.get(api);
```
Comment (JantuDeb) [inline src/context/product-listing-context.js:18]: const {data}= await axios.get(api);

**Review by JantuDeb — APPROVED** [review]
Suggested some good coding practices.

File: src/components/categorybanner/CategoryBanner.js [hunk 7]
```diff
@@ -1,34 +1,37 @@
...
-            <img className="baner-image" src="./assets/baner.png" alt="baner-book" />
+          <img
+            className="baner-image"
+            src="./assets/baner.png"
+            alt="baner-book"
+          />
```
Comment (mariyasada) [inline src/components/categorybanner/CategoryBanner.js:32]: use Link

**Review by mariyasada — APPROVED** [review]
Like the code structure 
Use Link instead of a tag in some parts

**Pull request merged**